site/en/guides/porting/set-up-the-build-environment.md

```diff
@@ -1,122 +1,143 @@
-name.
+> Note: For all path and code examples in this porting guide, always replace
+`{platform-name}` with the name of your new platform example. Most of the
+examples in the guide use a `efr32` platform name.
 
-## Step 1: GNU Autoconf
+[cmake-homepage]: https://cmake.org/
 
-The [Autoconf](https://www.gnu.org/software/autoconf/autoconf.html) script
-contains the basic system configuration options, including specific
-platform-relative macro definitions. These macros can be exposed for
-conditional compilation in other Makefiles during the pre-compiling phase.
+## Step 1: Create a new repository
 
-The OpenThread Autoconf script is located at:
-[`/openthread/configure.ac`](https://github.com/openthread/openthread/blob/main/configure.ac)
+The first step is set up a new home for your hardware platform. In this guide,
+we'll be creating a new repo named `ot-efr32` which will house the
+platform-abstraction layer, the hardware platform's SDK, and a few useful
+scripts.
 
-### Platform example name
+> *TODO* : Do we want to demonstrate making a new git repo through GitHub?
 
-In the `AC_ARG_WITH(examples ...)` macro, add the new hardware platform example name. The name
-should be added in alphabetical order.
 
-Example:
+```shell
+~$ mkdir -p ~/repos
+~$ cd ~/repos
+~/repos$ git clone git@github.com:SiliconLabs/ot-efr32.git
+Cloning into 'ot-efr32'...
+remote: Enumerating objects: 99, done.
+remote: Counting objects: 100% (99/99), done.
+remote: Compressing objects: 100% (60/60), done.
+remote: Total 333 (delta 65), reused 39 (delta 39), pack-reused 234
+Receiving objects: 100% (333/333), 170.78 KiB | 5.69 MiB/s, done.
+Resolving deltas: 100% (194/194), done.
 
-```
-AC_ARG_WITH(examples,
-    [AS_HELP_STRING([--with-examples=TARGET],
-        [Specify the examples from one of: none, simulation, cc2538, cc2650, efr32, nrf52840 @&lt;:@default=none@:&gt;@.])],
-    [
-        case "${with_examples}" in 
-        none)
-            ;;
-        simulation|cc2538|cc2650|efr32|nrf52840)
-            if test ${enable_posix_app} = "yes"; then
-                AC_MSG_ERROR([--with-examples must be none when POSIX apps are enabled by --enable-posix-app])
-            fi
-            ;;
-        *)
-            AC_MSG_ERROR([Invalid value ${with_examples} for --with-examples])
-            ;;
-        esac
-    ],
-    [with_examples=none])
+~/repos/ot-efr32$ git status
+On branch main
+Your branch is up to date with 'origin/main'.
+
+nothing to commit, working tree clean
 ```
 
-### Platform-specific C preprocessor symbol
+## Step 2: Download boiler-plate files
 
-Define a platform-specific C preprocessor symbol for the platform example and
-expose it.
+> *TODO* : Should we create a template repo that contains everything needed for porting? I'm thinking it would include a stubbed out PAL, stubbed out READMEs, CMake files, scripts, etc. This would be a good way to keep the structure of these new repos relatively consistent with the existing repos.
```

Comment:
Do we want to provide rough guidelines for structuring the repo? This might help maintain consistency across vendor repos if they ever want to have a fork of their new repo on the `openthread` GitHub organization. If so, what's the best way to do so?

Reply:
I think we can just list the top-level directories/files.